content/examples/02-next-pages/05-chat/01-generate-chat-completion.mdx

```diff
@@ -0,0 +1,102 @@
+---
+title: Generation Chat Completion
+description: Learn to generate chat completions using the Vercel AI SDK in your Next.js pages router application
+---
+
+# Generate Chat Completion
+
+Previously, you were able to generate text and objects using either a single message prompt, a system prompt, or a combination of both of them. However, there may be times when you want to generate text based on a series of messages.
+
+A chat completion allows you to generate text based on a series of messages. This series of messages can be any series of interactions between any number systems, but the most popular and relatable use case has been a series of messages that represent a conversation between a user and a model.
+
+<Browser>
+  <ChatGeneration
+    history={[
+      { role: 'User', content: 'How is it going?' },
+      { role: 'Assistant', content: 'All good, how may I help you?' },
+    ]}
+    inputMessage={{ role: 'User', content: 'Why is the sky blue?' }}
+    outputMessage={{
+      role: 'Assistant',
+      content: 'The sky is blue because of rayleigh scattering.',
+    }}
+  />
+</Browser>
+
+## Client
+
+Let's start by creating a simple chat interface with an input field that sends the user's message and displays the conversation history. You will call the `/api/generate-chat` endpoint to generate the assistant's response.
+
+```tsx filename='pages/index.tsx'
+import { CoreMessage } from 'ai';
+import { useState } from 'react';
+
+export default function Page() {
+  const [input, setInput] = useState('');
+  const [messages, setMessages] = useState<CoreMessage[]>([]);
+
+  return (
+    <div>
+      <input
+        value={input}
+        onChange={event => {
+          setInput(event.target.value);
+        }}
+        onKeyDown={async event => {
+          if (event.key === 'Enter') {
+            setMessages([...messages, { role: 'user', content: input }]);
+
+            await fetch('/api/chat', {
+              method: 'POST',
+              body: JSON.stringify({
+                messages: [...messages, { role: 'user', content: input }],
+              }),
+            }).then(async response => {
```

Comment:
combining `await` and `then` seems off. Can you change to async/await here? `const response = await fetch ...`

Reply:
done!